aws-analytics-pinpoint/src/androidTest/java/com/amplifyframework/analytics/pinpoint/AnalyticsPinpointInstrumentedTest.java

```diff
@@ -191,6 +200,43 @@ public void testUnregisterGlobalProperties() throws AnalyticsException {
         assertFalse(analyticsClient.getAllEvents().get(0).has("attributes"));
     }
 
+    @Test
+    public void testIdentifyUser() {
+        // Create a location object
+        AnalyticsUserProfile.Location location = new AnalyticsUserProfile.Location(SEATTLE_LATITUDE,
+                SEATTLE_LONGITUDE,
+                "98122",
+                "Seattle",
+                "WA",
+                "USA");
+        PinpointProperties pinpointProperties = PinpointProperties.builder()
+                .add("TestStringProperty", "TestStringValue")
+                .add("TestDoubleProperty", 1.0)
+                .build();
+        AnalyticsUserProfile userProfile = new AnalyticsUserProfile("test-user",
+                "user@test.com",
+                "test-plan",
+                location,
+                pinpointProperties);
+
+        Amplify.Analytics.identifyUser("userId", userProfile);
+
+        // Test endpoint attributes and metrics.
+        EndpointProfile endpointProfile = targetingClient.currentEndpoint();
+        EndpointProfileLocation endpointProfileLocation = endpointProfile.getLocation();
+        assertEquals(endpointProfile.getAttribute("email").get(0), "user@test.com");
+        assertEquals(endpointProfile.getAttribute("name").get(0), "test-user");
+        assertEquals(endpointProfile.getAttribute("plan").get(0), "test-plan");
+        assertEquals(endpointProfileLocation.getLatitude(), (Double) SEATTLE_LATITUDE);
+        assertEquals(endpointProfileLocation.getLongitude(), (Double) SEATTLE_LONGITUDE);
+        assertEquals(endpointProfileLocation.getPostalCode(), "98122");
+        assertEquals(endpointProfileLocation.getCity(), "Seattle");
+        assertEquals(endpointProfileLocation.getRegion(), "WA");
+        assertEquals(endpointProfileLocation.getCountry(), "USA");
+        assertEquals(endpointProfile.getAttribute("TestStringProperty").get(0), "TestStringValue");
+        assertEquals(endpointProfile.getMetric("TestDoubleProperty"), (Double) 1.0);
```

Comment:
Once you get `equals()` working correctly on these POJOs, you want to end up with something like a _single_ `assertEquals()` call, on the top-level object:
```
assertEquals(expectedComplexType, actualComplexType);
```

Internally, `assertEquals` will call `equals()` on the object. And _those_ should cascade through the fields, determining equality.

Or, closer to what you'll actually have:
```
assertEquals(
    // Expected data. Build one here, inline, that contains the field values you _expect_ to see.
    EndpoingProfile.builder()
        .addAttribute("name", Collections.singletonList("test-user"))
        .addAttribute("...", "... etc. "'")
       .build(),
    // The actual expirementally obtained copy ...
   endpointProfile
);
```

Also, as a note - in your `assertEquals(expected, actual)` calls, you currently have the expected and actual positions backwards. (Expected data should be first position, you have it as second position.) This makes a difference when the test fails, and jUnit goes to print out the error. It ordinarily would say something like "Expected \"Seatte\", but got \"Toledo\"." Yours would end up saying something like "Expected \"Toledo\", but got \"Seattle\"."